src/NServiceBus.RavenDB/Outbox/OutboxPersister.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace NServiceBus.RavenDB.Outbox
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using NServiceBus.Outbox;
+    using NServiceBus.RavenDB.Persistence;
+    using Raven.Client;
+
+    class OutboxPersister : IOutboxStorage
+    {
+        readonly ISessionProvider sessionProvider;
+
+        public OutboxPersister(ISessionProvider sessionProvider)
+        {
+            this.sessionProvider = sessionProvider;
+        }
+
+        // TODO this is a friction point, as it potentially allows for TryGet and SetAsDispatched to work
+        // TODO against a different DocumentStore than the one sessionProvider is working against
```

Comment:
To be honest I don't see the friction point. @synhershko any pointers? Because it works...